### PR TITLE
Improvements to the `PlotGridConfiguration` class

### LINKED ISSUE #274: Implement `print` for `PlotGridConfiguration`
With some meaningful content.

### LINKED ISSUE #294: Add fields to `PlotGridConfiguration` class to control tag aesthetics

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
- [x] Closes #274 
- [x] Closes #294

### CODE REVIEW

**Comment by IndrajeetPatil** [comment]
Print output:

``` r
library(tlf)

set.seed(123)
ls_plots <- list(
  # first plot
  plotBoxWhisker(mtcars,
    dataMapping = BoxWhiskerDataMapping$new(x = "am", y = "wt"), outliers = FALSE
  ),
  # second plot
  plotBoxWhisker(ToothGrowth,
    dataMapping = BoxWhiskerDataMapping$new(x = "supp", y = "len")
  )
)

plotGridObj <- PlotGridConfiguration$new(ls_plots)

plotGridObj$title <- "my combined plot"
plotGridObj$subtitle <- "something clever"
plotGridObj$caption <- "something dumb"
plotGridObj$nColumns <- 2L
plotGridObj$nRows <- 1L
plotGridObj$byRow <- TRUE
plotGridObj$tagLevels <- "A"
plotGridObj$tagPrefix <- "Plot ("
plotGridObj$tagSuffix <- ")"

plotGridObj
#> PlotGridConfiguration: 
#>    Plot grid annotations: 
#>      Title: my combined plot 
#>      Subtitle: something clever 
#>      Caption: something dumb 
#>    Plot grid arrangement: 
#>      Number of plots included: 2 
#>      Number of columns in the grid: 2 
#>      Number of rows in the grid: 1 
#>      Arranged in row-major order: TRUE 
#>    Individual plot tags: 
#>      Tag level format: A 
#>      Tag level prefix: Plot ( 
#>      Tag level suffix: ) 
#>      Tag level separator:
```

<sup>Created on 2022-05-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

**Comment by IndrajeetPatil** [comment]
Converting to draft because I also want to handle #294 in this PR.

File: DESCRIPTION [hunk 1]
```diff
@@ -1,93 +1,98 @@
...
-    'utilities-export.R'
-    'utilities-label.R'
-    'utilities-legend.R'
-    'utilities-mapping.R'
-    'utilities-theme.R'
-    'utils.R'
+Package: tlf
+Type: Package
+Title: TLF Library
+Version: 1.4.0
```
Comment (IndrajeetPatil) [inline DESCRIPTION:4]: @Yuri05 Bumping the version. Is that okay?

File: appveyor.yml [hunk 2]
```diff
@@ -12,10 +12,10 @@ install:
   - git submodule update --init --recursive
 
 environment:
-  app_version: "1.3"
+  app_version: "1.4"
```
Comment (IndrajeetPatil) [inline appveyor.yml:15]: @Yuri05 Same as DESCRIPTION

**Review by PavelBal — APPROVED** [review]

File: DESCRIPTION (hunk 1 above)
Comment (Yuri05) [reply DESCRIPTION:4]: Yes :)

File: R/plot-grid.R [hunk 3]
```diff
@@ -132,23 +187,89 @@ plotGrid <- function(plotGridConfiguration) {
...
   "PlotGridConfiguration",
+  inherit = ospsuite.utils::Printable,
   public = list(
     plotList = list(),
+
+    # title ------------------------------------
+
     title = NULL,
+    titleColor = "black",
+    titleSize = 14,
```
Comment (Yuri05) [inline R/plot-grid.R:242]: don't we have this (and other sizes below) as a constant somewhere in the tlf?

Comment (IndrajeetPatil) [reply R/plot-grid.R]: No, but I have created an enum for it now.

File: R/aaa-enums.R [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+# R6 class generators will create classes at build time. This means that if any
+# of the enums are to be used inside these class environments, they need to be
+# already available in the NAMESPACE when classes are being created. If not, you
+# will get "object not found" error because R6 class generator can't find these
+# enums in the NAMESPACE.
+#
+# This whole problem can be easily avoided by collating NAMESPACE alphabetically
+# and including the enums in `aaa-` file which will be loaded before all other
+# files and therefore all R6 class environments can find the needed enums since
+# these classes will be defined in these other files.
```
Comment (IndrajeetPatil) [inline R/aaa-enums.R:15]: @pchelle Please make note of this. I have moved all enums to a single file now.

I need to some more research into this, but keeping them in a single file is a good solution at the moment.

**Comment by IndrajeetPatil** [comment]
@Yuri05 I have made the changes you had asked. Let me know if you want me to change anything else.

**Review by Yuri05 — APPROVED** [review]

**Comment by Yuri05** [comment]
>Let me know if you want me to change anything else.

Change the world :)
PR looks good, merging

**Pull request merged**